dts/bindings/adc/adc-controller.yaml

```diff
@@ -12,3 +13,129 @@ properties:
     "#io-channel-cells":
       type: int
       required: true
+
+    "#address-cells":
+      required: false
+      const: 1
+
+    "#size-cells":
+      required: false
+      const: 0
+
+child-binding:
```

Comment:
Several of these do not appear in https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/iio/adc/adc.yaml. That in itself is not a problem, but I am concerned about whether some of them are Zephyr-API-specific. If any of these properties are specific to the Zephyr ADC API, we should prefix their names with `zephyr,` as discussed here:

https://lists.zephyrproject.org/g/devel/message/8456

Reply:
Adjusted, although it looks a bit odd perhaps as I could re-use only the `reg` and `bipolar` properties from the Linux binding.